### DISCUSSION TITLE
RDKit Dependency OS Error

### DISCUSSION THREAD

**Discussion by feaidaina** [rdkit/rdkit#4023] [Q&A]
We are conducting a study on chemical shift prediction via machine learning and one of the libraries involved is the RDKit. We are currently basing it on [this model](https://git.rwth-aachen.de/avt.svt/public/graph_neural_network_for_fuel_ignition_quality) that utilizes graph neural networks.

With using such, we encountered an error in the dependency [BaseFeature.fdef](https://ip-147-251-124-124.flt.cloud.muni.cz/kratochvil/rdkit/-/blob/aefdd50def9cc2366276a8510c1d2e704b609684/Data/BaseFeatures.fdef). We would like to ask about the possible ways in order to alleviate this error.
![image](https://user-images.githubusercontent.com/82097609/114357774-ba736a80-9ba4-11eb-8cc5-20d88251b789.png)

**Comment by greglandrum** [comment]
Hi @feaidaina : that file is normally created when you `conda install` the RDKit.
In order to try and identify the problem, please share how you installed the RDKit as well as the actual code that is producing the error (not just the error message).

**Reply by feaidaina** [reply]
This was how I installed conda and rdkit.
![image](https://user-images.githubusercontent.com/82097609/114362767-4936b600-9baa-11eb-8f36-844bd769d209.png)

For the part of the code that causes the error, it's rooted from here 
```
dataset = FUELNUMBERS(
    path + 'Train/',
    pre_transform=MyPreTransform(),
    pre_filter=MyFilter())
```
![image](https://user-images.githubusercontent.com/82097609/114363216-c3673a80-9baa-11eb-8a86-848fc197ae56.png)
![image](https://user-images.githubusercontent.com/82097609/114363352-e1cd3600-9baa-11eb-8ee4-2f5c09aa3c60.png)

**Reply by greglandrum** [reply]
Thanks for the detail!
It looks like the problem here is that you are installing the RDKit using pip and the kora package and I guess that isn't correctly installing the RDKit - specifically it doesn't seem to be installing the data files which the RDKit needs.

The best solution I can think of to this is to install the RDKit using conda instead of pip/kora. I don't use google colab much, but it looks like this may be a way for you to do this: https://stackoverflow.com/a/63158487

**Reply by feaidaina** [reply]
Hello! I tried the most upvoted answer and I dont know why this error is coming up
![image](https://user-images.githubusercontent.com/82097609/114504914-2d8ee680-9c62-11eb-8596-2218e4bbcc2f.png)

**Reply by greglandrum** [reply]
Ok, I did a bit of investigating and this turns out to be non-trivial. Fortunately @jaimergp has already done all the work and made things easy!
The README in this repo shows what you need to do:
https://github.com/jaimergp/condacolab
and here's a simple demo notebook I just used to verify that it works:
https://colab.research.google.com/drive/1eSQbCXnvo3VGOFYH8JMUmzuXoTA1VScV?usp=sharing

**Comment by CKannas** [comment]
Hi,

I've found this discussion
https://www.javaer101.com/en/article/45136767.html

Best regards,

Christos

Christos Kannas

**Comment by feaidaina** [comment]
This happens when i do either of the options mentioned on the link
![image](https://user-images.githubusercontent.com/82097609/114544344-14e7f600-9c8d-11eb-95d1-8ff2e9226801.png)
